vmdk_plugin/drivers/photon/photon_driver.go

```diff
@@ -618,6 +633,16 @@ func (d *VolumeDriver) Unmount(r volume.UnmountRequest) volume.Response {
 		return volume.Response{Err: ""}
 	}
 
+	if plugin_utils.IsFullVolumeName(r.Name) != true {
+		fullVolName := fullVolumeNameMap[r.ID]
+		if fullVolName == "" { // if ID not present in local map, do a get trip
+			status, _ := d.GetVolume(r.Name)
+			datastore := plugin_utils.GetDatastore(r.Name, status)
+			fullVolName = plugin_utils.GetFullVolumeName(r.Name, datastore)
+		}
+		r.Name = fullVolName
+	}
```

Comment:
I suggest to wrap up L639->644 (with slight modification) in `COnstructFullVolumeName (shortName string) string ` function returning full name (and error) . Otherwise cut-n-paste is on the way